docs/pages/1 - Intro to Mill.md

```diff
@@ -701,9 +701,16 @@ mill mill.scalalib.Dependency/showUpdates
 mill mill.scalalib.Dependency/showUpdates --allowPreRelease true # also show pre-release versions
 ```
 
-## IntelliJ Support
+## IDE Support
 
-Mill supports IntelliJ by default. Use `mill mill.scalalib.GenIdea/idea` to
+Mill supports any IDE that is compatible with [BSP](https://build-server-protocol.github.io/), such as IntelliJ, VSCode
```

Comment:
So, I've just tried to use the latest master (0.8.0-9-6a7a11) with metals, ie removed all pre-existing bloop configuration / metals configuration, generated bsp config from the contrib bsp module, and opened vscode (waiting for it to connect to the bsp server). I get the following logs  : 

```
2020.10.05 10:34:50 INFO  logging to file /private/tmp/foo/.metals/metals.log
2020.10.05 10:34:50 INFO  started: Metals version 0.9.4 in workspace '/private/tmp/foo' for client vscode.
2020.10.05 10:34:51 INFO  time: initialize in 0.38s
Initializing Scala Debugger
2020.10.05 10:34:51 INFO  tracing is enabled: /Users/oliviermelois/Library/Caches/org.scalameta.metals/bsp.trace.json
2020.10.05 10:34:51 INFO  skipping build import with status 'Dismissed'
2020.10.05 10:34:53 INFO  no build target: using presentation compiler with only scala-library
2020.10.05 10:34:53 WARN  no build target for: /private/tmp/foo/build.sc
2020.10.05 10:34:54 INFO  time: connected to build server in 2.58s
2020.10.05 10:34:54 INFO  Connected to Build server v0.8.0-9-6a7a11
2020.10.05 10:34:55 INFO  time: imported build in 0.77s
2020.10.05 10:34:54 INFO  disconnected: build server
2020.10.05 10:34:54 ERROR Failed to connect with build server, no functionality will work.
java.lang.NullPointerException
	at scala.math.Ordering$StringOrdering.compare(Ordering.scala:333)
	at scala.math.Ordering$StringOrdering.compare$(Ordering.scala:333)
	at scala.math.Ordering$String$.compare(Ordering.scala:335)
	at scala.math.Ordering$String$.compare(Ordering.scala:335)
	at scala.math.Ordering$$anon$1.compare(Ordering.scala:122)
	at java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
	at java.util.TimSort.sort(TimSort.java:220)
	at java.util.Arrays.sort(Arrays.java:1438)
	at scala.collection.SeqLike.sorted(SeqLike.scala:659)
	at scala.collection.SeqLike.sorted$(SeqLike.scala:647)
	at scala.collection.AbstractSeq.sorted(Seq.scala:45)
	at scala.collection.SeqLike.sortBy(SeqLike.scala:634)
	at scala.collection.SeqLike.sortBy$(SeqLike.scala:634)
	at scala.collection.AbstractSeq.sortBy(Seq.scala:45)
	at scala.meta.internal.metals.Doctor.buildTargetRows(Doctor.scala:391)
	at scala.meta.internal.metals.Doctor.$anonfun$buildTargetsTable$22(Doctor.scala:382)
	at scala.meta.internal.metals.Doctor.$anonfun$buildTargetsTable$22$adapted(Doctor.scala:382)
	at scala.meta.internal.metals.HtmlBuilder.element(HtmlBuilder.scala:97)
	at scala.meta.internal.metals.Doctor.$anonfun$buildTargetsTable$12(Doctor.scala:382)
	at scala.meta.internal.metals.Doctor.$anonfun$buildTargetsTable$12$adapted(Doctor.scala:382)
	at scala.meta.internal.metals.HtmlBuilder.element(HtmlBuilder.scala:97)
	at scala.meta.internal.metals.Doctor.buildTargetsTable(Doctor.scala:382)
	at scala.meta.internal.metals.Doctor.$anonfun$buildTargetsHtml$2(Doctor.scala:287)
	at scala.meta.internal.metals.Doctor.$anonfun$buildTargetsHtml$2$adapted(Doctor.scala:287)
	at scala.meta.internal.metals.HtmlBuilder.call(HtmlBuilder.scala:76)
	at scala.meta.internal.metals.Doctor.buildTargetsHtml(Doctor.scala:287)
	at scala.meta.internal.metals.Doctor.executeDoctor(Doctor.scala:97)
	at scala.meta.internal.metals.Doctor.executeRefreshDoctor(Doctor.scala:82)
	at scala.meta.internal.metals.Doctor.executeReloadDoctor(Doctor.scala:76)
	at scala.meta.internal.metals.Doctor.check(Doctor.scala:123)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$connectToNewBuildServer$8(MetalsLanguageServer.scala:1751)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$indexWorkspace$2(MetalsLanguageServer.scala:1942)
	at scala.meta.internal.metals.MetalsLanguageServer.timedThunk(MetalsLanguageServer.scala:1849)
	at scala.meta.internal.metals.MetalsLanguageServer.indexWorkspace(MetalsLanguageServer.scala:1925)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$profiledIndexWorkspace$2(MetalsLanguageServer.scala:1876)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.meta.internal.metals.MetalsLanguageServer.timedThunk(MetalsLanguageServer.scala:1849)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$profiledIndexWorkspace$1(MetalsLanguageServer.scala:1876)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

I'm afraid saying it works with vscode is a bit premature :/

Reply:
I should remove the mention of VSCode because they rely on Bloop instead.

Reply:
> I should remove the mention of VSCode because they rely on Bloop instead.

Not necessarily : if the bsp config exists, metals will try to connect to it (after the user prevents it from loading it via bloop config). 

But yeah, I think the safest route for now is to just make the documentation/wording specifically mention Intellij

Reply:
Btw, this issue you are seeing here was actually a Metals issue, not a Mill issue @Baccata. I've corrected it here: https://github.com/scalameta/metals/pull/2164.

If you'd like to try again with Metals, the latest snapshot has this fix.